### PR TITLE
Fixed issue #17846: Expression manager on file upload question type don't work on same page

### PR DESCRIPTION
**Pull request by gabrieljenik**
When rendering file upload question types, file upload hidden fields are not following usual naming conventions as other fields.
Its processing then was skipped.

Now, looking field name on dictionary.

### CODE REVIEW

**Comment by Shnoulle** [comment]
Checking this issue : there are more error on concept …

I think we must fix naming conventions (`answerSGQA` and `javaSGQA`)

**Comment by Shnoulle** [comment]
here you add a javascript test on ALL page for an issue in input id convention …

We already have long JS issue with array and array of checkbox, adding more js must really be avoided if we can.

**Comment by gabrieljenik** [comment]
You have a point.
I am in doubt in between the balance of risk vs effort vs scope...

Let me review...

**Comment by Shnoulle** [comment]
Adding java in id for html and a $("#java{SGQ}").[trigger](https://github.com/LimeSurvey/LimeSurvey/blob/8fdbfdabd6be1c29725766aa684f1f5881805d2e/assets/packages/expressions/em_javascript.js#L238) ?
Remind the _filecount please too.

Touching HTML of one question type or Expression_Manager used on all question type ? Risky one is ?

**Comment by gabrieljenik** [comment]
> Adding java in id for html 

This inplies not only updating the rendering of the FileUpload question but also changing:
- JS related to that file upload
- PHP processing the field
- Testing all that!
- Backward compatibility with every custom plugin anyone could have
- Backward compatibilty with child question themes (guessing not many)

I wouldn't do it if I would have to decide

**Comment by glimz** [comment]
I agree with @Shnoulle, we should not tweak but provide a major issue.

**Review by Shnoulle — COMMENTED** [review]
Move to trigger('updated') ?

File: assets/scripts/expressions/em_javascript.js [hunk 1]
```diff
@@ -46,6 +46,10 @@ $(document).on("change",".radio-item :radio:not([onclick]), .button-item :radio:
 $(document).on("change",".checkbox-item :checkbox:not([onclick]),.button-item :checkbox:not([onclick])",function(event){
     checkconditions($(this).val(), $(this).attr('name'), 'checkbox', 'click')
 });
+/* file upload */
+$(document).on("change",".upload-item input[id^='java']",function(event){
```
[the remark this answers is not in the document]
Comment (gabrieljenik) [reply assets/scripts/expressions/em_javascript.js:50]: > `.upload-item :hidden` ?

Do we need the :hidden selector?

[the remark this answers is not in the document]
Comment (Shnoulle) [reply assets/scripts/expressions/em_javascript.js:50]: I don't like the start selector : it was time consuming.

For :hidden : we already use it for equation 

If you update event : it work with current js i think.

```
$(document).on("updated",".answer-item :hidden",function(event){
    checkconditions($(this).val(), $(this).attr('name'), 'equation', 'updated')
});
```

It's not really a request change ;) . More : less cod eline is (maybe) better.

File: assets/scripts/expressions/em_javascript.js [hunk 2]
```diff
@@ -46,9 +46,9 @@ $(document).on("change",".radio-item :radio:not([onclick]), .button-item :radio:
 $(document).on("change",".checkbox-item :checkbox:not([onclick]),.button-item :checkbox:not([onclick])",function(event){
     checkconditions($(this).val(), $(this).attr('name'), 'checkbox', 'click')
 });
-/* file upload */
-$(document).on("change",".upload-item input[id^='java']",function(event){
-    checkconditions($(this).val(), $(this).attr('name'), 'upload', 'change')
+/* hidden item */
+$(document).on("updated",".answer-item :hidden, .upload-item :hidden",function(event){
```
Comment (Shnoulle) [inline assets/scripts/expressions/em_javascript.js:50]: `{{ coreClass }}` didn't include answer-item ?

**Review by Shnoulle — APPROVED** [review]
answer-item msut be added according to : https://manual.limesurvey.org/Theme_editor#Global_classes_for_question_part

But it's another (minor issue)

**Pull request merged**